### PR TITLE
Decompose command options

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr

### CODE REVIEW

File: cmd/bbolt/command_surgery_cobra.go [hunk 1]
```diff
@@ -44,8 +32,23 @@ func newSurgeryCobraCommand() *cobra.Command {
...
+func (o targetDBOptions) Validate() error {
+	if o.filePath == "" {
+		return fmt.Errorf("output database path wasn't given, specify output database file path with --output option")
+	}
+	return nil
+}
+
 func newSurgeryRevertMetaPageCommand() *cobra.Command {
-	cfg := defaultSurgeryOptions()
+	o := targetDBOptions{}
```
Comment (ahrtr) [inline cmd/bbolt/command_surgery_cobra.go:51]: nit: Golang recommends `var o targetDBOptions`

File: cmd/bbolt/command_surgery_cobra.go [hunk 2]
```diff
@@ -44,8 +32,23 @@ func newSurgeryCobraCommand() *cobra.Command {
 	return surgeryCmd
 }
 
+type targetDBOptions struct {
+	filePath string
+}
```
Comment (ahrtr) [inline cmd/bbolt/command_surgery_cobra.go:37]:
```suggestion
type surgeryBaseOptions struct {
	outputDBFilePath string
}
```

Comment (serathius) [reply cmd/bbolt/command_surgery_cobra.go:37]: Good suggestion

File: cmd/bbolt/command_surgery_cobra.go (hunk 1 above)
Comment (serathius) [reply cmd/bbolt/command_surgery_cobra.go]: Done

File: cmd/bbolt/command_surgery_cobra.go [hunk 3]
```diff
@@ -134,12 +155,36 @@ func surgeryCopyPageFunc(srcDBPath string, cfg surgeryOptions) error {
...
-	fmt.Fprintf(os.Stdout, "The page %d was successfully copied to page %d\n", cfg.surgerySourcePageId, cfg.surgeryDestinationPageId)
+	fmt.Fprintf(os.Stdout, "The page %d was successfully copied to page %d\n", cfg.sourcePageId, cfg.destinationPageId)
+	return nil
+}
+
+type clearPageElementsOptions struct {
+	surgeryBaseOptions
+	pageId         uint64
+	startElementId int
+	endElementId   int
```
Comment (ahrtr) [inline cmd/bbolt/command_surgery_cobra.go:166]: nit: `startElementIdx` or `startElementIndex` is better than `startElementId`. There is no a concept `element ID`, and it isn't global unique either.
```suggestion
	startElementIdx int
	endElementIdx   int
```

File: cmd/bbolt/command_surgery_cobra.go [hunk 4]
```diff
@@ -134,12 +155,36 @@ func surgeryCopyPageFunc(srcDBPath string, cfg surgeryOptions) error {
 		fmt.Fprintf(os.Stdout, "Please consider executing `./bbolt surgery abandon-freelist ...`\n")
 	}
 
-	fmt.Fprintf(os.Stdout, "The page %d was successfully copied to page %d\n", cfg.surgerySourcePageId, cfg.surgeryDestinationPageId)
+	fmt.Fprintf(os.Stdout, "The page %d was successfully copied to page %d\n", cfg.sourcePageId, cfg.destinationPageId)
+	return nil
+}
+
+type clearPageElementsOptions struct {
```
Comment (ahrtr) [inline cmd/bbolt/command_surgery_cobra.go:162]:
```suggestion
type surgeryClearPageElementsOptions struct {
```

File: cmd/bbolt/command_surgery_cobra.go [hunk 5]
```diff
@@ -86,8 +90,30 @@ func surgeryRevertMetaPageFunc(srcDBPath string, cfg surgeryOptions) error {
 	return nil
 }
 
+type copyPageOptions struct {
```
Comment (ahrtr) [inline cmd/bbolt/command_surgery_cobra.go:93]:
```suggestion
type surgeryCopyPageOptions struct {
```

File: cmd/bbolt/command_surgery_cobra.go (hunk 3 above)
Comment (serathius) [reply cmd/bbolt/command_surgery_cobra.go:166]: Done

File: cmd/bbolt/command_surgery_cobra.go (hunk 4 above)
Comment (serathius) [reply cmd/bbolt/command_surgery_cobra.go:162]: Done

File: cmd/bbolt/command_surgery_cobra.go (hunk 5 above)
Comment (serathius) [reply cmd/bbolt/command_surgery_cobra.go:93]: Done

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @serathius

**Pull request merged**